test/onnx/test_op_consistancy.py

```diff
@@ -0,0 +1,460 @@
+        if len(module_name) == 2:
+            # Remove the torch. prefix
+            op_name = f"{module_name[1]}.{op_name}"
+        return op_name
+    # Already a string
+    return op
+
+
+def opsets_before(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is before the specified."""
+
+    def compare(other_opset: int):
+        return other_opset < opset
+
+    return compare
+
+
+def opsets_after(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is after the specified."""
+
+    def compare(other_opset: int):
+        return other_opset > opset
+
+    return compare
+
+
+def reason_onnx_runtime_does_not_support(
+    operator: str, dtypes: Optional[Sequence[str]] = None
+) -> str:
+    """Formats the reason: ONNX Runtime doesn't support the given dtypes."""
+    return f"{operator} on {dtypes or 'dtypes'} not supported by ONNX Runtime"
+
+
+def reason_onnx_does_not_support(
+    operator: str, dtypes: Optional[Sequence[str]] = None
+) -> str:
+    """Formats the reason: ONNX doesn't support the given dtypes."""
+    return f"{operator} on {dtypes or 'dtypes'} not supported by the ONNX Spec"
+
+
+def reason_jit_tracer_error(info: str) -> str:
+    """Formats the reason: JIT tracer errors."""
+    return f"JIT tracer error on {info}"
+
+
+def reason_flaky() -> str:
+    """Formats the reason: test is flaky."""
+    return "flaky test"
+
+
+# Modify this section ##########################################################
+# NOTE: Modify this section as more ops are supported. The list should be sorted
+# alphabetically.
+#
+# For example, to add a test for torch.ceil:
+# 1.  Add `torch.ceil` to ALLOWLIST_OP then run pytest.
+#         You can also add a string, e.g. "ceil" or "__radd__".
+# 2a. If the test fails, fix the error or add a new entry to EXPECTED_SKIPS_OR_FAILS.
+# 2b. If the test is expected to fail only on certain opsets, add a new entry to
+#     EXPECTED_OPSET_FAILS.
+
+# Ops to be tested for consistency between onnx and pytorch
+ALLOWLIST_OP: AbstractSet[str] = frozenset(
+    map(
+        get_torch_op_name,
+        (
+            torch.ceil,
+            torch.sqrt,
+            torch.t,
+        ),
+    )
+)
+
+# fmt: off
+# Turn off black formatting to keep the list compact
+
+# Expected failures for onnx export. If an op is expected to fail only for certain
+# ONNX opsets, add the op to EXPECTED_OPSET_FAILS below.
+# The list should be sorted alphabetically by op name.
+# Q: When should I use skip vs vs dont_care vs xfail?
+# A: Use skip when we want to fix the test eventually but it doesn't fail consistently,
+#        e.g. the test is flaky. Otherwise, use xfail.
+#    Use dont_care if we don't care about the test passing, e.g. ONNX doesn't support the usage.
+#    Use xfail if a test fails now and we want to eventually fix the test.
+EXPECTED_SKIPS_OR_FAILS: Tuple[DecorateMeta, ...] = (
```

Comment:
I'm not sure if it's worth adding all this bespoke architecture to support skips and xfails on these ONNX tests. Why not just add DecorateInfos to the OpInfos as necessary? They seem comparably complicated